src/storage/wal_replay.cpp

```diff
@@ -403,47 +404,86 @@ void ReplayState::ReplayDropTableMacro(BinaryDeserializer &deserializer) {
 // Replay Index
 //===--------------------------------------------------------------------===//
 void ReplayState::ReplayCreateIndex(BinaryDeserializer &deserializer) {
-	auto info = deserializer.ReadProperty<unique_ptr<CreateInfo>>(101, "index");
+
+	auto create_info = deserializer.ReadProperty<unique_ptr<CreateInfo>>(101, "index_catalog_entry");
+	auto index_info = deserializer.ReadProperty<IndexStorageInfo>(102, "index_storage_info");
+	D_ASSERT(index_info.IsValid() && !index_info.name.empty());
+
+	auto &storage_manager = db.GetStorageManager();
+	auto &single_file_sm = storage_manager.Cast<SingleFileStorageManager>();
+	auto &block_manager = single_file_sm.block_manager;
+	auto &buffer_manager = block_manager->buffer_manager;
+
+	deserializer.ReadList(103, "index_storage", [&](Deserializer::List &list, idx_t i) {
```

Comment:
Do we need to do the `ReadList` here?

Reply:
In `write_ahead_log.cpp`, we write `allocation_size` bytes at the `buffer_ptr` position. The list that we serialize there, in `SerializeIndexToWAL`, contains all ART blocks from which we have to write data.

Here, we need to read these bytes for each list entry. I don't know how to generate code for that. As I can tell the serializer to write the struct, but not to write `allocation_size` bytes at `buffer_ptr`.

Relevant code:
```cpp
	//! Contains all buffer pointers and their allocation size for serializing to the WAL
	//! First dimension: all fixed-size allocators, second dimension: the buffers of each allocator
	vector<vector<IndexBufferInfo>> buffers;

//! Information to serialize an index buffer to the WAL
struct IndexBufferInfo {
	IndexBufferInfo(data_ptr_t buffer_ptr, const idx_t allocation_size) :
 		buffer_ptr(buffer_ptr), allocation_size(allocation_size) {}

	data_ptr_t buffer_ptr;
	idx_t allocation_size;
};
```